core/src/main/java/org/apache/gravitino/authorization/AuthorizationUtils.java

```diff
@@ -364,4 +376,127 @@ private static void checkCatalogType(
+                      Catalog catalogObj =
+                          GravitinoEnv.getInstance().catalogDispatcher().loadCatalog(identifier);
+                      if (catalogObj.provider().equals("hive")) {
+                        Schema schema =
+                            GravitinoEnv.getInstance()
+                                .schemaDispatcher()
+                                .loadSchema(
+                                    NameIdentifier.of(
+                                        metalake,
+                                        catalogObj.name(),
+                                        "default" /*Hive default schema*/));
+                        if (schema.properties().containsKey(HiveConstants.LOCATION)) {
+                          String defaultSchemaLocation =
+                              schema.properties().get(HiveConstants.LOCATION);
+                          if (defaultSchemaLocation != null && !defaultSchemaLocation.isEmpty()) {
+                            locations.add(defaultSchemaLocation);
+                          } else {
+                            LOG.warn("Catalog %s location is not found", ident);
+                          }
+                        }
+                      }
+                    });
+          }
+          break;
+        case CATALOG:
+          {
+            Catalog catalogObj = GravitinoEnv.getInstance().catalogDispatcher().loadCatalog(ident);
+            if (catalogObj.provider().equals("hive")) {
+              Schema schema =
+                  GravitinoEnv.getInstance()
+                      .schemaDispatcher()
+                      .loadSchema(
+                          NameIdentifier.of(
+                              metalake, catalogObj.name(), "default" /*Hive default schema*/));
+              if (schema.properties().containsKey(HiveConstants.LOCATION)) {
+                String defaultSchemaLocation = schema.properties().get(HiveConstants.LOCATION);
+                if (defaultSchemaLocation != null && !defaultSchemaLocation.isEmpty()) {
+                  locations.add(defaultSchemaLocation);
+                } else {
+                  LOG.warn("Catalog %s location is not found", ident);
+                }
+              }
+            }
+          }
+          break;
+        case SCHEMA:
+          {
+            Schema schema = GravitinoEnv.getInstance().schemaDispatcher().loadSchema(ident);
+            if (schema.properties().containsKey(HiveConstants.LOCATION)) {
+              String schemaLocation = schema.properties().get(HiveConstants.LOCATION);
+              if (schemaLocation != null && schemaLocation.isEmpty()) {
+                locations.add(schemaLocation);
+              } else {
+                LOG.warn("Schema %s location is not found", ident);
+              }
+            }
+          }
+          break;
+        case TABLE:
+          {
+            Table table = GravitinoEnv.getInstance().tableDispatcher().loadTable(ident);
```

Comment:
`getInstance().tableDispatcher()` contains too many unnesseary dispatchers, what you need is `TableOperationDispatcher` YES?

Reply:
Currently, Gravitino didn't implement get TableOperationDispatcher in the GravitinoEnv.